File: db4oj.tests/src/com/db4o/db4ounit/common/ta/ta/TADateTestCase.java
/* Copyright (C) 2007  Versant Inc.  http://www.db4o.com */

package com.db4o.db4ounit.common.ta.ta;

import java.util.*;

import db4ounit.*;

public class TADateTestCase extends TAItemTestCaseBase {

    public static Date first = new Date(1195401600000L);

    public static void main(String[] args) {
        new TADateTestCase().runAll();
    }

    protected void assertItemValue(Object obj) throws Exception {
        TADateItem item = (TADateItem) obj;
        Assert.areEqual(first, item.getUntyped());
        Assert.areEqual(first, item.getTyped());
    }

    protected void assertRetrievedItem(Object obj) throws Exception {
        TADateItem item = (TADateItem) obj;
        Assert.isNull(item._untyped);
        Assert.areEqual(emptyValue(),item._typed);
    }
    
    private Object emptyValue() {
        return db().reflector().forClass(Date.class).nullValue();
    }

    protected Object createItem() throws Exception {
        TADateItem item = new TADateItem();
        item._typed = first;
        item._untyped = first;
        return item;
    }
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/TA/TA/TADateTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4oUnit;
using Db4objects.Db4o.Tests.Common.TA.TA;

namespace Db4objects.Db4o.Tests.Common.TA.TA
{
	public class TADateTestCase : TAItemTestCaseBase
	{
		public static DateTime first = new DateTime(1195401600000L);

		public static void Main(string[] args)
		{
			new TADateTestCase().RunAll();
		}

		/// <exception cref="System.Exception"></exception>
		protected override void AssertItemValue(object obj)
		{
			TADateItem item = (TADateItem)obj;
			Assert.AreEqual(first, item.GetUntyped());
			Assert.AreEqual(first, item.GetTyped());
		}

		/// <exception cref="System.Exception"></exception>
		protected override void AssertRetrievedItem(object obj)
		{
			TADateItem item = (TADateItem)obj;
			Assert.IsNull(item._untyped);
			Assert.AreEqual(EmptyValue(), item._typed);
		}

		private object EmptyValue()
		{
			return Db().Reflector().ForClass(typeof(DateTime)).NullValue();
		}

		/// <exception cref="System.Exception"></exception>
		protected override object CreateItem()
		{
			TADateItem item = new TADateItem();
			item._typed = first;
			item._untyped = first;
			return item;
		}
	}
}
